### ISSUE TITLE
request for complete source archive through github releases 

### ISSUE THREAD

**Issue by aryalaadi** [zed-industries/zed#8816]
### Check for existing issues

- [X] Completed

### Describe the feature

providing a complete source archive would be very convenient for distributors and packagers, compared to having git as a middleman to initiate submodules 
i think there is a way to automate this with github workflows to archive and publish the complete sources on every release

**Comment by Moshyfawn** [comment]
Isn't this what you're looking for?

<img width="1284" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/60a6acaa-c6fc-4065-b291-520df7b6c6da">

**Comment by aryalaadi** [comment]
> Isn't this what you're looking for?

No. the default github release does not include the submodules, so you cant build it, the archive is practically useless.
The archive is source-only and has no git artifacts, so you cant initialize the sub modules conveniently.
heres an example of something i want, [this project](https://github.com/taisei-project/taisei/releases/tag/v1.3) provides the source archives with the sub modules and tells package maintainers to ignore the default github build
![Screenshot_2024-03-04_21-07-39](https://github.com/zed-industries/zed/assets/141743392/e212d016-83df-48dc-abaa-86dc13fa4acb)

**Comment by Moshyfawn** [comment]
Ah, I see where you're coming from #8779.
